src/nvim/testdir/test_startup.vim

```diff
@@ -499,6 +499,20 @@ func Test_read_stdin()
   call delete('Xtestout')
 endfunc
 
+func Test_set_shell()
+  let after = [
+	\ 'call writefile([&shell], "Xtestout")',
+	\ 'quit!',
+	\ ]
+  let $SHELL = '/bin/with space/sh'
+  if RunVimPiped([], after, '', '')
+    let lines = readfile('Xtestout')
+    " MS-Windows adds a space after the word
+    call assert_equal('/bin/with\ space/sh', lines[0])
```

Comment:
Doesn't this fail with the rest of the patch?

Reply:
Try running the tests with `SHELL` having a space. Then, try it on Windows. The patch should have used double quotes instead in line with the suggestion in `:h shell`.